pkg/controllers/vmsetcontroller/vmsetcontroller.go

```diff
@@ -225,6 +226,10 @@ func (v *VirtualMachineSetController) reconcileVirtualMachineSet(vmset *hfv1.Vir
 		glog.V(5).Infof("provisioning %d vms", needed)
 		// this code is so... verbose...
 		for i := 0; i < needed; i++ {
+			sshUser, exists := env.Spec.TemplateMapping[vmt.Name]["ssh_username"]
```

Comment:
This change will only set the sshUser when a VMSet is created, which breaks existing behavior. I think this type of behavior should be moved/duplicated to a level closer to the VM, so that existing VMs are not suddenly broken.

Reply:
Would those existing VMs be broken if the default username is still "ubuntu"?

Reply:
Theoretically no, if you were able to set the existing VMs (or default in the code somewhere) then they would be fine. I am also thinking of use cases where VMs are created outside of using a VMset, like dynamic provisioning and so forth

Reply:
Makes sense. The SshUsername will need to be set on the VM object whenever a VM itself is created, which in current config is either via the vmset controller or the dynamic bind controller. Does it make sense to add this functionality to the dynamic bind controller and leave this here? Or else how would one go about setting the username on the VM resource? Set it post-creation via an independent VM controller of some kind?

Reply:
Honestly, I'm not opposed to behavior where it just defaults to `ubuntu` and if you want anything else you need to explicitly set it yourself

Reply:
I think that's what we have now (except I need to implement that in the DBC, my oversight). 
Or are you saying don't bother with this bit here in the vmset controller _at all_, and let the shell handle the default vs. vmspec behavior?

Reply:
Having this functionality both here and in the DBC would for sure work. 

The shell should be able to handle the "catch-all" cases i.e. where a VM template does not have a defined ssh username, a VM is created out of band of VMSet (by some other unknown controller), or a VM was created pre-definition of ssh_username

Reply:
Ok. I will add this to the DBC and push for your review!